cub/cub/device/device_reduce.cuh

```diff
@@ -102,6 +169,119 @@ struct unzip_and_write_arg_extremum_op
+            typename NumItemsT>
+  CUB_RUNTIME_FUNCTION static cudaError_t rfa_reduce_impl(
+    void* d_temp_storage,
+    size_t& temp_storage_bytes,
+    InputIteratorT d_in,
+    OutputIteratorT d_out,
+    NumItemsT num_items,
+    T init,
+    cudaStream_t stream,
+    _CUDA_VSTD::false_type /* is_supported */);
+
+  template <typename TuningEnvT, typename InputIteratorT, typename OutputIteratorT, typename T, typename NumItemsT>
+  CUB_RUNTIME_FUNCTION static cudaError_t rfa_reduce_impl(
+    void* d_temp_storage,
+    size_t& temp_storage_bytes,
+    InputIteratorT d_in,
+    OutputIteratorT d_out,
+    NumItemsT num_items,
+    T init,
+    cudaStream_t stream,
+    _CUDA_VSTD::true_type /* is_supported */)
+  {
+    using reduction_op_t  = ::cuda::std::plus<>;
+    using offset_t        = detail::choose_offset_t<NumItemsT>;
+    using accum_t         = ::cuda::std::__accumulator_t<reduction_op_t, detail::it_value_t<InputIteratorT>, T>;
+    using transform_t     = ::cuda::std::__identity;
+    using reduce_tuning_t = ::cuda::std::execution::
+      __query_result_or_t<TuningEnvT, detail::reduce::get_reduce_tuning_query_t, detail::reduce::default_rfa_tuning>;
+    using policy_t = typename reduce_tuning_t::template type<accum_t, offset_t, reduction_op_t>;
+    using dispatch_t =
+      detail::DispatchReduceDeterministic<InputIteratorT, OutputIteratorT, offset_t, T, accum_t, transform_t, policy_t>;
+
+    return dispatch_t::Dispatch(
+      d_temp_storage, temp_storage_bytes, d_in, d_out, static_cast<offset_t>(num_items), init, stream);
+  }
+
+  template <typename TuningEnvT,
+            typename InputIteratorT,
+            typename OutputIteratorT,
+            typename ReductionOpT,
+            typename T,
+            typename NumItemsT>
+  CUB_RUNTIME_FUNCTION static cudaError_t reduce_impl(
+    void* d_temp_storage,
+    size_t& temp_storage_bytes,
+    InputIteratorT d_in,
+    OutputIteratorT d_out,
+    NumItemsT num_items,
+    ReductionOpT,
+    T init,
+    ::cuda::execution::determinism::gpu_to_gpu_t,
+    cudaStream_t stream)
+  {
+    using offset_t    = detail::choose_offset_t<NumItemsT>;
+    using accum_t     = ::cuda::std::__accumulator_t<ReductionOpT, detail::it_value_t<InputIteratorT>, T>;
+    using transform_t = ::cuda::std::__identity;
+
+    // RFA is only supported for float and double accumulators
+    constexpr bool is_float_or_double = _CUDA_VSTD::is_same_v<accum_t, float> || _CUDA_VSTD::is_same_v<accum_t, double>;
+    constexpr bool is_sum             = _CUDA_VSTD::is_same_v<ReductionOpT, ::cuda::std::plus<>>;
+    constexpr bool is_supported       = is_float_or_double && is_sum;
+
+    static_assert(is_supported, "gpu-to-gpu deterministic reduction supports only float and double sum.");
+
+    return rfa_reduce_impl<TuningEnvT>(
+      d_temp_storage,
+      temp_storage_bytes,
+      d_in,
+      d_out,
+      num_items,
+      init,
+      stream,
+      _CUDA_VSTD::bool_constant<is_supported>{});
```

Comment:
```suggestion
    if constexpr(is_supported) {
    return rfa_reduce_impl<TuningEnvT>(
      d_temp_storage,
      temp_storage_bytes,
      d_in,
      d_out,
      num_items,
      init,
      stream);
    }
```
And remove the overload of `rfa_reduce_impl` with `_CUDA_VSTD::false_type` as last parameter.